model/src/main/kotlin/AnalyzerResult.kt

```diff
@@ -46,8 +47,14 @@ data class AnalyzerResult(
         /**
          * The list of all errors.
          */
-        val errors: SortedMap<Identifier, List<Error>>
-) : CustomData() {
+        val errors: SortedMap<Identifier, List<Error>>,
+
+        /**
+         * A map that holds arbitrary data. Can be used by third-party tools to add custom data to the model.
+         */
+        @JsonInclude(JsonInclude.Include.NON_EMPTY)
+        val data: Map<String, Any> = emptyMap()
```

Comment:
Should we introduce a `typealias CustomData = Map<String, Any>`?

Reply:
Not sure, I like typealiases to reduce repetition in a single file, but in this case I think it would make the code less explicit and harder to read. But I'm open to be convinced of the opposite ;)

Reply:
I find a common `typealias` a bit more speaking to emphasize "it's the same thing" across all these classes that `data` is meant to be used for. And if we'd ever want to change how custom data is stored, we'd only need to change it in a single place.

Reply:
Ok, I'd like to this in a separate PR then, to unblock another change which depends on this PR.